### PR TITLE
[WIP] Make actuator to work without cluster object

### PR DESCRIPTION
**Pull request by vikaschoudhary16**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```
/cc @derekwaynecarr @enxebre @ingvagabund

### CODE REVIEW

File: pkg/apis/awsprovider/v1alpha1/awsmachineproviderconfig_types.go [hunk 1]
```diff
@@ -72,8 +85,34 @@ type AWSMachineProviderSpec struct {
...
+
+	// LoadBalancers is the set of load balancers to which the new instance
+	// should be added once it is created.
+	LoadBalancers []LoadBalancerReference `json:"loadBalancers"`
 }
 
+// Placement indicates where to create the instance in AWS
+type Placement struct {
+	// Region is the region to use to create the instance
+	Region string `json:"region"`
```
Comment (derekwaynecarr) [inline pkg/apis/awsprovider/v1alpha1/awsmachineproviderconfig_types.go]: i think this field can be dropped, the cluster is scoped to a region already.

**Comment by vincepri** [comment]
Can we add some extended description with use cases and issues that this PR solves?

File: pkg/apis/awsprovider/v1alpha1/awsmachineproviderconfig_types.go (hunk 1 above)
Comment (derekwaynecarr) [reply pkg/apis/awsprovider/v1alpha1/awsmachineproviderconfig_types.go]: Per our discussion, I think API proposal should probably be a separate PR.

Comment (vikaschoudhary16) [reply pkg/apis/awsprovider/v1alpha1/awsmachineproviderconfig_types.go]: @derekwaynecarr https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/508

**Comment by detiber** [comment]
Closing pending a design proposal

/close

**Pull request closed without merging**